### PR TITLE
WT-10578 Create ARM64/x86_64 Memory Model Test tool

### PR DESCRIPTION
**Pull request by jeremythorp**
This work created a tool that can perform a series of tests of operations, some of which allow memory accesses to be made out of order, and some which do not.

The test runs on both x86_64 and ARM64, and specifically includes tests that have operations that cannot generate out of order memory access on x86_64, but can on ARM64.

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -3211,6 +3211,19 @@ tasks:
...
+    commands:
+      - func: "get project"
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/tools/memory-model-test"
+          script: |
+            set -o errexit
+            set -o verbose
+            export "PATH=/opt/mongodbtoolchain/v4/bin:$PATH"
+            g++ -o memory_model_test -O2 memory_model_test.cpp -lpthread -std=c++20
```
Comment (korteland) [inline test/evergreen.yml:3224]: Should we ask for `-Wall -Werror` both here and in `CMakeLists.txt`?

File: tools/memory-model-test/README.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+* C++ compiler that supports C++ 20
+* (optional) CMake and ninja
+
+### Build
+
+There are two options:
+
+* Use the CMakeLists.txt file with CMake:
+  ```
+  md build
```
Comment (korteland) [inline tools/memory-model-test/README.md:44]: I think `md` is a Jeremy's shell specific command ;-)

File: tools/memory-model-test/README.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+## The Tests
+
+Memory Model Test runs a series of tests. Each test has two threads which are accessing shared variables. 
+
+All shared variables are set to 0 before each test.
+
+There are two broad groups of tests:
+- Group 1: 
+  - Each thread writes 1 to a shared variable (`x` for thread 1, and `y` for thead 2),
+    and then reads from the other shared variable (`y` for thread 1, and `x` for thead 2)
```
Comment (korteland) [inline tools/memory-model-test/README.md:22]: typos: `thead` -> `thread`

**Review by korteland — COMMENTED** [review]
Couple of smaller points - looks good otherwise.

File: tools/memory-model-test/README.md (hunk 3 above)
Comment (jeremythorp) [reply tools/memory-model-test/README.md]: Fixed!

File: tools/memory-model-test/README.md (hunk 2 above)
Comment (jeremythorp) [reply tools/memory-model-test/README.md]: Well spotted - now fixed.

File: test/evergreen.yml (hunk 1 above)
Comment (jeremythorp) [reply test/evergreen.yml]: Great idea - now added.

**Review by korteland — APPROVED** [review]
LGTM!

**Pull request merged**